components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle

```
@@ -1,69 +1,160 @@
+        return
+    }
+
+    def enableGradleEnterprise = Boolean.getBoolean("com.gradle.enterprise.build_validation.enableGradleEnterprise")
+
+    def gePluginVersion = enableGradleEnterprise ? "3.12.1" : null
+    def ccudPluginVersion = enableGradleEnterprise ? "1.8.2" : null
 
     repositories {
         gradlePluginPortal()
     }
 
     dependencies {
-        classpath("com.gradle:gradle-enterprise-gradle-plugin:${gradleEnterprisePluginVersion}")
+        if (gePluginVersion) {
+            classpath "com.gradle:gradle-enterprise-gradle-plugin:$gePluginVersion"
+        }
+
+        if (ccudPluginVersion) {
+            classpath "com.gradle:common-custom-user-data-gradle-plugin:$ccudPluginVersion"
+        }
     }
 }
 
-// Don't run against the included builds (if the main build has any).
-def isTopLevelBuild = gradle.getParent() == null
-if (isTopLevelBuild) {
-    def version = GradleVersion.current().baseVersion
-    def atLeastGradle5 = version >= GradleVersion.version("5.0")
-    def atLeastGradle6 = version >= GradleVersion.version("6.0")
-    def isScanDump = System.properties.containsKey("scan.dump")
+def BUILD_SCAN_PLUGIN_ID = 'com.gradle.build-scan'
+def BUILD_SCAN_PLUGIN_CLASS = 'com.gradle.scan.plugin.BuildScanPlugin'
+
+def GRADLE_ENTERPRISE_PLUGIN_ID = 'com.gradle.enterprise'
+def GRADLE_ENTERPRISE_PLUGIN_CLASS = 'com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin'
+def GRADLE_ENTERPRISE_EXTENSION_CLASS = 'com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension'
+
+def CCUD_PLUGIN_ID = 'com.gradle.common-custom-user-data-gradle-plugin'
+def CCUD_PLUGIN_CLASS = 'com.gradle.CommonCustomUserDataGradlePlugin'
+
+def isTopLevelBuild = !gradle.parent
+if (!isTopLevelBuild) {
+    return
+}
+
+def enableGradleEnterprise = Boolean.getBoolean("com.gradle.enterprise.build_validation.enableGradleEnterprise")
+def isScanDump = System.properties.containsKey("scan.dump")
+
+clearWarnings()
+
+// define a buildScanPublished listener that captures the build scan data and appends to a file
+def buildScanPublishedAction = { def buildScan ->
+    if (buildScan.metaClass.respondsTo(buildScan, 'buildScanPublished', Action)) {
+        def scanFile = new File(experimentDir, "build-scans.csv")
+        buildScan.buildScanPublished { publishedBuildScan ->
+            def buildScanUri = publishedBuildScan.buildScanUri
+            def buildScanId = publishedBuildScan.buildScanId
+            def port = (buildScanUri.port != -1) ? ":" + buildScanUri.port : ""
+            def baseUrl = "${buildScanUri.scheme}://${buildScanUri.host}${port}"
+            scanFile.append("${baseUrl},${buildScanUri},${buildScanId}\n")
+        }
+        def errorFile = new File(experimentDir, "build-scan-publish-error.txt")
+        buildScan.onError { error ->
+            errorFile.text = error
+        }
+    }
+}
 
-    clearWarnings()
+// register buildScanPublished listener and optionally apply the GE / Build Scan plugin
```

-1: I don't see a buildScanPublished listener being registered below anywhere.

```suggestion
// Optionally apply the GE / Build Scan plugin
```

---

The listener is being applied [here](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/a2e2bf985267d51e5b2c59502c72bba7da28e811/components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle#L107) and [here](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/a2e2bf985267d51e5b2c59502c72bba7da28e811/components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle#L139).

---

So it is! 👍